expression/builtin_cast_test.go

```diff
@@ -81,6 +81,15 @@ func (s *testEvaluatorSuite) TestCast(c *C) {
 		res, err = f.Eval(nil)
 		c.Assert(err, IsNil)
 	}
+
+	// cast(1234 as char(0))
+	tp.Flen = 0
+	tp.Charset = charset.CharsetUTF8
+	f = NewCastFunc(tp, &Constant{Value: types.NewDatum(1234), RetType: types.NewFieldType(mysql.TypeString)}, ctx)
+	res, err = f.Eval(nil)
+	c.Assert(err, IsNil)
```

Comment:
There will be a warning in MySQL:
mysql> SELECT 1, CAST(1234 AS CHAR(0));
+---+-----------------------+
| 1 | CAST(1234 AS CHAR(0)) |
+---+-----------------------+
| 1 |                       |
+---+-----------------------+
1 row in set, 1 warning (0.00 sec)